include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
@jfirebaugh This does not work as hoped. Since the `Layer::Impl` subclasses no longer contain a reference to the cascaded/evaluated paint properties, the render layers created in the `GeometryTileWorker` start with a blank copy, disjointed from the copy in the render layers in the style.